scripts/travis/install-requirements.sh

```diff
@@ -1,8 +1,13 @@
 #!/usr/bin/env bash
 
-if [ "${TASK}" = 'compilepy3 ci-py3-unit' ] || [ "${TASK}" = 'ci-py3-integration' ]; then
+if [[ "${TASK}" = *'ci-py3'* ]]; then
     pip install "tox==3.8.6"
 
+    # cleanup any invalid python2 cache
+    test -d virtualenv/lib/${PYTHON_VERSION} || rm -rf virtualenv/*
```

Comment:
Can you please provide more context around this cleanup?

Are we removing something that should be cached by Travis and saved build time before?

Reply:
I discovered that the python3 builds sometimes ran with a cached python2 virtualenv, invalidating the tests. I added this fix in c15112b81514c79be3fbb2cc17ae1e3bbda34898.

The makefile only checks to see if the activate script is present, not if it is using the right version of python, so if the cache does inadvertently get corrupted (ie as I experienced with a cache that included python2 instead of python3) I want to catch that and force rebuilding the virtualenv.

This is somewhat related to the change that catches `*ci-py3*` tasks. I want to make sure this runs for all python3 tasks.

Reply:
Same here - add comments to this effect so the next person who reads this understands why it's important.

Reply:
added a comment.